### ISSUE TITLE
License

### ISSUE THREAD

**Issue by ScruffR** [BulldogLowell/LiquidCrystal_I2C_Spark#1]
Hi James,
have you actually built this library from scratch?
If not it would be good and just fair to give credit to the original author.
Additionally it would help users to access any help the original author may provide.

**Comment by BulldogLowell** [comment]
Yes, you are right I should show where it came from.  I did reference its origins in the Spark forum.

it was created by the author named "Anonymous" 

https://gist.github.com/anonymous/dc7ecc5e6574105b4fbd

**Comment by ScruffR** [comment]
For me it looks very much like this 
 https://github.com/fdebrabander/Arduino-LiquidCrystal-I2C-library
which gives credits to
 http://www.dfrobot.com/

**Comment by BulldogLowell** [comment]
Right, perhaps you should notify Anonymous to give credit to http://www.dfrobot.com/

I am not here and I have never taken credit for someone else's work, if that is what you are implying.

The link to the Anonymous Git was in the Spark Forum long before I ever was.

**Comment by ScruffR** [comment]
I'm not implying anything, expecially not that you intended to take credit for other peoples work. 
I'd rather commend you for bringing this to the Particle family.

But the thing with "Anonymous" is that it is not the name of a github user, but the default branch where gists get stored if you don't have an account or just don't sign in.
So it was either the original uploader of the gist who forgot to provide the source or it was given in a thread which originally refered to that gist, but the context got lost.

Sorry, if you got the impression I was pointing fingers :blush: 

---

After some backtracking I found the first occurence of this gist on the forum here
 https://community.particle.io/t/lcd-i2c-lib-info-customcharacters/4670
